### PR TITLE
Fix debug info

### PR DESCRIPTION
**Pull request by kito-cheng**
https://github.com/riscv/riscv-gnu-toolchain/issues/219

### CODE REVIEW

File: gas/config/tc-riscv.c [hunk 1]
```diff
@@ -1932,30 +1933,31 @@ md_apply_fix (fixS *fixP, valueT *valP, segT seg ATTRIBUTE_UNUSED)
 
 	    case BFD_RELOC_RISCV_CFA:
 	      /* Load the byte to get the subtype.  */
-	      subtype = bfd_get_8 (NULL, &fixP->fx_frag->fr_literal[fixP->fx_where]);
+	      subtype = bfd_get_8 (NULL, &((fragS *) (fixP->fx_frag->fr_opcode))->fr_literal[fixP->fx_where]);
+	      loc = fixP->fx_frag->fr_fix - (subtype & 7);
```
Comment (aswaterman) [inline gas/config/tc-riscv.c]: offsetT instead of long?

File: gas/config/tc-riscv.c [hunk 2]
```diff
@@ -1847,6 +1847,7 @@ md_apply_fix (fixS *fixP, valueT *valP, segT seg ATTRIBUTE_UNUSED)
   unsigned int subtype;
   bfd_byte *buf = (bfd_byte *) (fixP->fx_frag->fr_literal + fixP->fx_where);
   bfd_boolean relaxable = FALSE;
+  long loc;
```
Comment (aswaterman) [inline gas/config/tc-riscv.c]: offsetT instead of long?

**Comment by aswaterman** [comment]
@palmer-dabbelt want to try to sneak this upstream?

**Comment by kito-cheng** [comment]
updated, use offsetT instead of long.

**Comment by palmer-dabbelt** [comment]
Thanks for fixing this.  It looks like I missed the 2.28 release window, so I'm just going to submit it to master.

**Pull request merged**